deeplake/integrations/mmdet/mmdet_utils.py

```diff
@@ -0,0 +1,499 @@
+                        np.uint8
+                    )
+                    if masks is not None
+                    else None,  # optimize here
+                    "iscrowd": int(is_crowds[bbox_index]),
+                }
+
+                imgToAnns[row_index].append(ann)
+                anns[absolute_id] = ann
+                absolute_id += 1
+
+        category_names = self.class_names  # TO DO: add super category names
+        category_names = [
+            {"id": cat_id, "name": name} for cat_id, name in enumerate(category_names)
+        ]
+
+        for idx, category_name in enumerate(category_names):
+            cats[idx] = category_name
+
+        for ann in anns.values():
+            catToImgs[ann["category_id"]].append(ann["image_id"])
+
+        # create class members
+        self.anns = anns
+        self.imgToAnns = imgToAnns
+        self.catToImgs = catToImgs
+        self.imgs = imgs
+        self.cats = cats
+        print("create index done!")
+
+    def getAnnIds(self, imgIds=[], catIds=[], areaRng=[], iscrowd=None):
+        """
+        Get ann ids that satisfy given filter conditions. default skips that filter
+        :param imgIds  (int array)     : get anns for given imgs
+               catIds  (int array)     : get anns for given cats
+               areaRng (float array)   : get anns for given area range (e.g. [0 inf])
+               iscrowd (boolean)       : get anns for given crowd label (False or True)
+        :return: ids (int array)       : integer array of ann ids
+        """
+        imgIds = imgIds if _isArrayLike(imgIds) else [imgIds]
+        catIds = catIds if _isArrayLike(catIds) else [catIds]
+
+        if len(imgIds) == len(catIds) == len(areaRng) == 0:
+            # anns = self.dataset['annotations']
+            anns = list(self.anns.values())
+        else:
+            if not len(imgIds) == 0:
+                lists = [
+                    self.imgToAnns[imgId] for imgId in imgIds if imgId in self.imgToAnns
+                ]
+                anns = list(itertools.chain.from_iterable(lists))
+            else:
+                # anns = self.dataset['annotations']
+                anns = list(self.anns.values())
+            anns = (
+                anns
+                if len(catIds) == 0
+                else [ann for ann in anns if ann["category_id"] in catIds]
+            )
+            anns = (
+                anns
+                if len(areaRng) == 0
+                else [
+                    ann
+                    for ann in anns
+                    if ann["area"] > areaRng[0] and ann["area"] < areaRng[1]
+                ]
+            )
+        if not iscrowd == None:
+            ids = [ann["id"] for ann in anns.values() if ann["iscrowd"] == iscrowd]
+        else:
+            ids = [ann["id"] for ann in anns]
+        return ids
+
+    def getCatIds(self, catNms=[], supNms=[], catIds=[]):
+        """
+        filtering parameters. default skips that filter.
```

Comment:
can we format these docstrings the way it's been done in the rest of the codebase?